components/feature/push/src/main/java/mozilla/components/feature/push/SubscriptionWorker.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.feature.push
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequest
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import kotlinx.coroutines.Dispatchers
+import mozilla.components.concept.push.PushProcessor
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * A [WorkManager] worker that will check if push subscriptions are valid.
+ *
+ * This worker can be run every cold-start of the app but the verification will only take place
+ * after [PERIODIC_INTERVAL_MILLISECONDS]. Be sure to also consider the limitation of WorkManager.
+ */
+internal open class SubscriptionWorker(private val context: Context, params: WorkerParameters) :
+    CoroutineWorker(context, params) {
+
+    private val logger = Logger("SubscriptionWorker")
+    private var lastVerified: Long
+        get() = context.getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)
```

Comment:
nit: We could improve the readability if we use the same functions helper as you did in the test class `preferences` and `updatePref`


```kotlin
    companion object {
        private fun preferences(context: Context): SharedPreferences =
            context.getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)

        private fun updatePref(value: Long) {
            return preferences(testContext)
                .edit()
                .putLong(SubscriptionWorker.LAST_VERIFIED, value)
                .apply()
        }
}
```

Reply:
I was trying to reduce that old way since I needed to read and write to the pref, but in the tests I just needed to write. :)

Maybe I should use a custom accessor for the tests as well to be consistent.